### PR TITLE
fix onednn ConvTranspose2d channels last issue when ic=1

### LINKED ISSUE #99519: [PT2.0] Channels_last for weight for ConvTranpose gives Random output
### 🐛 Describe the bug

If weight tensor's memory format changed to channels_last for ConvTranspose2d Operator then we get the random output
Please use below code to reproduce issue in PT 2.0

```
import torch

ifm = torch.ones([1, 1, 2, 2]).contiguous(memory_format=torch.channels_last)
model = torch.nn.ConvTranspose2d(in_channels=1, out_channels=2, kernel_size=[5, 5])
model.weight.data = torch.ones([1, 2, 5, 5]).contiguous(
    memory_format=torch.channels_last
)
fwd_out = model(ifm)
print(fwd_out)
```

We get Output as:
```
tensor([[[[5.3065e-34, 3.7275e-43,        nan, 7.3318e+22, 1.1141e+27,
           3.4590e-12],
          [1.1446e+24, 1.1028e+21, 2.4471e+32, 2.6219e+20, 1.8990e+28,
           7.1447e+31],
          [1.7753e+28, 1.7034e+19, 3.1839e-15, 4.4849e+21, 6.4640e-04,
           9.0941e-04],
          [1.7701e+31, 2.8930e+12, 7.1758e+22, 2.0704e-19, 1.6532e+19,
           1.4353e-19],
          [7.5338e+28, 1.7753e+28, 1.7034e+19, 5.2165e-11, 1.8888e+31,
           3.2745e-12],
          [1.9051e+31, 4.1579e+21, 1.4583e-19, 1.9421e+20, 1.4353e-19,
           7.5338e+28]],

         [[0.0000e+00, 0.0000e+00, 0.0000e+00, 4.5972e+24, 1.9618e+20,
           2.0572e+26],
          [9.1042e-12, 7.2065e+31, 2.7374e+20, 2.1127e-19, 2.9602e+29,
           2.2266e-15],
          [1.8151e+25, 2.8973e+23, 7.3988e+31, 2.7370e+20, 1.8461e+20,
           1.8939e+34],
          [2.7208e+23, 7.5338e+28, 1.4603e-19, 4.4907e+21, 7.2060e+31,
           2.7530e+12],
          [1.3556e-19, 1.8151e+25, 2.8973e+23, 1.3570e-14, 1.2712e+31,
           7.5555e+31],
          [6.9767e+22, 3.4452e-12, 1.8888e+31, 4.9567e+28, 2.7530e+12,
           7.6286e-19]]]], grad_fn=<ConvolutionBackward0>)
```
Excepted Output:
```
tensor([[[[0.8606, 1.8606, 1.8606, 1.8606, 1.8606, 0.8606],
          [1.8606, 3.8606, 3.8606, 3.8606, 3.8606, 1.8606],
          [1.8606, 3.8606, 3.8606, 3.8606, 3.8606, 1.8606],
          [1.8606, 3.8606, 3.8606, 3.8606, 3.8606, 1.8606],
          [1.8606, 3.8606, 3.8606, 3.8606, 3.8606, 1.8606],
          [0.8606, 1.8606, 1.8606, 1.8606, 1.8606, 0.8606]],

         [[1.0841, 2.0841, 2.0841, 2.0841, 2.0841, 1.0841],
          [2.0841, 4.0841, 4.0841, 4.0841, 4.0841, 2.0841],
          [2.0841, 4.0841, 4.0841, 4.0841, 4.0841, 2.0841],
          [2.0841, 4.0841, 4.0841, 4.0841, 4.0841, 2.0841],
          [2.0841, 4.0841, 4.0841, 4.0841, 4.0841, 2.0841],
          [1.0841, 2.0841, 2.0841, 2.0841, 2.0841, 1.0841]]]],
       grad_fn=<ConvolutionBackward0>
```

### Versions

Name: torch
Version: 2.0.0
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /home/jthakur/.pt_2_0/lib/python3.8/site-packages
Requires: filelock, jinja2, networkx, nvidia-cublas-cu11, nvidia-cuda-cupti-cu11, nvidia-cuda-nvrtc-cu11, nvidia-cuda-runtime-cu11, nvidia-cudnn-cu11, nvidia-cufft-cu11, nvidia-curand-cu11, nvidia-cusolver-cu11, nvidia-cusparse-cu11, nvidia-nccl-cu11, nvidia-nvtx-cu11, sympy, triton, typing-extensions
Required-by: torchaudio, torchvision, triton

cc @ezyang @gchanan @zou3519 @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jamesr66a @frank-wei @soumith @msaroufim @wconstab @ngimel @bdhirsh

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99539
* #99511

cc @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: test/test_mkldnn.py [hunk 1]
```diff
@@ -427,6 +427,17 @@ def test_conv_transpose2d(self):
...
+    def test_conv_transposed2d_ic1_nhwc(self):
+        x = torch.ones([1, 1, 2, 2]).contiguous(memory_format=torch.channels_last)
+        model = torch.nn.ConvTranspose2d(in_channels=1, out_channels=2, kernel_size=[5, 5]).eval()
+        model.weight.data = torch.ones([1, 2, 5, 5]).contiguous(memory_format=torch.channels_last)
+        with torch.no_grad():
+            y = model(x)
+            with torch.backends.mkldnn.flags(enabled=False):
+                y_ref = model(x)
+            self.assertEqual(y, y_ref)
+
```
Comment (malfet) [inline test/test_mkldnn.py:439]: Please use single empty line between two methods in class and two empty lines between top level functions/classes
```suggestion
```

Comment (XiaobingSuper) [reply test/test_mkldnn.py:439]: yes, changed.

**Review by mingfeima — APPROVED** [review]

**Comment by DanilBaibak** [comment]
@XiaobingSuper sorry, I have to revert your PR. It leads to internal failures. @mingfeima, @malfet could you pls help XiaobingSuper with fix.  Here you can find more details - [D45160029](https://www.internalfb.com/diff/D45160029).

**Comment by XiaobingSuper** [comment]
@DanilBaibak @malfet, I can't open [D45160029](https://www.internalfb.com/diff/D45160029). Could you help share the log or a reproduced code with me? Thanks!

**Comment by malfet** [comment]
> @DanilBaibak @malfet, I can't open [D45160029](https://www.internalfb.com/diff/D45160029). Could you help share the log or a reproduced code with me? Thanks!

Sure, the failure looked as follows:
```
[2023-04-20T13:53:30.219-07:00] Stderr: fbcode/caffe2/aten/src/ATen/native/mkldnn/Conv.cpp:711:5: error: no member named 'compute_v3' in 'ideep::convolution_transpose_forward'; did you mean 'ideep::convolution_forward::compute_v3'?
```

Was `convolution_transpose_forward::compute_v3` added to ideep/mkldnn as part of the same commit as `convolution_forward::compute_v3` or later? If later, than internal ideep is out of date again.

We need to agree on some cadence of those updates (as it don't have a bandwith to do it every week)

**Comment by XiaobingSuper** [comment]
> > @DanilBaibak @malfet, I can't open [D45160029](https://www.internalfb.com/diff/D45160029). Could you help share the log or a reproduced code with me? Thanks!
> 
> Sure, the failure looked as follows:
> 
> ```
> [2023-04-20T13:53:30.219-07:00] Stderr: fbcode/caffe2/aten/src/ATen/native/mkldnn/Conv.cpp:711:5: error: no member named 'compute_v3' in 'ideep::convolution_transpose_forward'; did you mean 'ideep::convolution_forward::compute_v3'?
> ```
> 
> Was `convolution_transpose_forward::compute_v3` added to ideep/mkldnn as part of the same commit as `convolution_forward::compute_v3` or later? If later, than internal ideep is out of date again.
> 
> We need to agree on some cadence of those updates (as it don't have a bandwith to do it every week)

it is added after ```convolution_forward::compute_v3```.

**Comment by malfet** [comment]
> > Was `convolution_transpose_forward::compute_v3` added to ideep/mkldnn as part of the same commit as `convolution_forward::compute_v3` or later? If later, than internal ideep is out of date again.
> > We need to agree on some cadence of those updates (as it don't have a bandwith to do it every week)
> 
> it is added after `convolution_forward::compute_v3`.

@XiaobingSuper do you know which commit added it?

**Comment by XiaobingSuper** [comment]
@malfet ,  ```convolution_transpose_forward::compute_v3``` is added by https://github.com/pytorch/pytorch/pull/97430, but https://github.com/pytorch/pytorch/pull/97157 do a wrong merge request which reverts the ideep update, then https://github.com/pytorch/pytorch/pull/98305 re-added the ideep update.

**Comment by XiaobingSuper** [comment]
Add @jgong5, we can talk about the update behavior at the next sync meeting.

**Pull request closed without merging**